src/components/Movie/MovieDetails.js

```diff
@@ -7,61 +9,42 @@ import SimilarMovies from './SimilarMovies';
 import MovieBanner from './MovieBanner';
 import './moviedetails.css';
 
+import * as movieActions from '../../actions/movieActions';
+
 require('es6-promise').polyfill();
 require('isomorphic-fetch');
 
-export default class MovieDetails extends Component {
+export class MovieDetails extends Component {
     constructor(props) {
         super(props);
-        this.state = {
-            movie: {},
-            selectedGenre: '',
-            similarMovies: []
-        };
     }
     componentDidMount() {
         let url = `${Constants.baseURL}/${this.props.match.params.id}`;
-        this.getRequest(url).then(response => {
-            this.populateState(response);
-            this.fetchSimilarMovies(response.genres[0]);
-        });
-    }
-    componentWillReceiveProps(nextProps) {
-        let url = `${Constants.baseURL}/${nextProps.match.params.id}`;
-        this.getRequest(url).then(response => {
-            this.populateState(response);
-            this.fetchSimilarMovies(response.genres[0]);
-        });
-    }
-    populateState(response) {
-        this.setState({
-            movie: response
-        });
-    }
-    fetchSimilarMovies(genre) {
-        let url = `${Constants.baseURL}?search=${genre}&searchBy=genres&sortBy=release_date&sortOrder=desc`;
-        this.getRequest(url)
-            .then(response => {
-                this.populateStateForSimilarMovies(genre, response.data);
-            });
-    }
-    getRequest(url) {
-        return fetch(url).then(res => res.json());
-    }
-    populateStateForSimilarMovies(genre, response) {
-        this.setState({
-            selectedGenre: genre,
-            similarMovies: response
-        });
+        this.props.actions.fetchMovieDetails(url);
     }
     render() {
         return (
             <div>
                 <Header showSearchButton={true} />
-                <MovieBanner movie={this.state.movie} />
-                <SimilarMovies genre={this.state.selectedGenre} movies={this.state.similarMovies} />
+                <MovieBanner movie={this.props.movie} />

```

Comment:
We have some important things that we can do here o/

Do you remember when we talked about components and containers?

A good practice in Redux world is have individual containers that know the domain that they are doing. As example in your case, you have MovieBanner and SimilarMovies, both receiveing props from this container.
But as you have a Single Source of Truth, you can create containers for this two components that are connected with Redux and they receive the props that they need. 

```javascript
render() {
....
<div>
    <Header showSearchButton />
    <MovieBanner />
    <SimilarMovies />
    <Footer />
</div>
...
}
```

Your code will look something like this. Also it's good to split containers and components into different directories.

Reply:
Added container components for a few files, due to lack of time.